File: zd_03.cpp
#include <iostream>
#include <cmath>
using namespace std;

const double H = 5.0;      // Max height in meters
const double L = 20.0;     // Horizontal range in meters
const double g = 9.81;     // Acceleration due to gravity

// Function: f(x) = 4H / L (constant)
double exmp_function(double x) {
    return (4 * H / L);
}

// Iterative method to solve for alpha
double iter_method(double x_0, double epsilon) {
    double x = x_0;
    int iter = 0;
    while (true) {
        double x_new = 4 * H / L;  // Fixed value
        if (abs(x_new - x) < epsilon) {
            cout << "Number of iterations: " << iter << endl;
            return x_new;
        }
        x = x_new;
        iter++;
    }
}

int main() {
    double x_0 = 0.5;          // Initial guess in radians
    double epsilon = 1e-8;     // Tolerance

    double x_root = iter_method(x_0, epsilon);
    cout << "Solving the equation: x = " << x_root << " radians ("
         << x_root * (180.0 / M_PI) << " degrees)" << endl;

    // Compute V using V = sqrt(gL / sin(2 * alpha))
    double V = sqrt(g * L / sin(2 * x_root));
    cout << "Initial velocity V = " << V << " m/s" << endl;

    return 0;
}

File: zd_03.py
from math import *

# Known values
H = 5.0   # Max height in meters
L = 20.0  # Horizontal range in meters
g = 9.81

# Function: f(x) = tan(a) - 4H / L
def exmp_function(x):
    return (4 * H / L)

# Iterative method to solve f(alpha) = 0
def iter_method(x_0, epsilon):
    x = x_0
    iter = 0
    while True:
        # Rearranged fixed-point form: alpha = arctan(4H / L)
        x_new = (4 * H / L)
        if abs(x_new - x) < epsilon:
            print(f"Number of iterations: {iter}")
            return x_new
        x = x_new
        iter += 1

# Initial guess and precision
x_0 = 0.5  # radians
epsilon = 1e-8

# Solving
x_root = iter_method(x_0, epsilon)
print(f"Solving the equation: x = {x_root} radians ({degrees(x_root)} degrees)")

# Now compute V using V = sqrt(gL / sin(2alpha))
V = sqrt(g * L / sin(2 * x_root))
print(f"Initial velocity V = {V:.4f} m/s")
